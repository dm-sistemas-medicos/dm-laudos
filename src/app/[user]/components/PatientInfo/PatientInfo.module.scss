.patient-info {
  padding: 4px;
  display: grid;
  grid-template-columns: 1fr auto auto;
  gap: 8px;
  font-size: 12px;
  border: solid #000 1px;

  > div {
    display: grid;
    grid-template-columns: auto 1fr;
    grid-auto-flow: column;
    align-items: center;
    column-gap: 2px;

    > input {
      font-size: 12px;
    }
  }
}
