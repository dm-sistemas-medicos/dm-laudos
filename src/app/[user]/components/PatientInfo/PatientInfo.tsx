"use client";

import { useState } from "react";

import TextInput from "./components/TextInput/TextInput";
import BirthdateInput from "./components/BirthdateInput/BirthdateInput";
import MedicalReportDateInput from "./components/MedicalReportDateInput/MedicalReportDateInput";

import users from "@/users";

import styles from "./PatientInfo.module.scss";

const today = new Date();

const PatientInfo = () => {
  const userInfo = users["mariana-machoski"];

  const [medicalReportDate, setMedicalReportDate] = useState(
    today.toISOString().substring(0, 10)
  );

  return (
    <form className={styles["patient-info"]}>
      <TextInput id="name" label="Nome" />
      <TextInput
        id="medical-record-number"
        label="Prontuário"
        isSmallInput={true}
      />
      <MedicalReportDateInput
        medicalReportDate={medicalReportDate}
        setMedicalReportDate={setMedicalReportDate}
      />
      <TextInput
        id="requester-name"
        label="Solicitante"
        initialInfo={userInfo.defaultRequester}
      />
      <BirthdateInput medicalReportDate={medicalReportDate} />
      <TextInput
        id="operator-name"
        label="Operador"
        initialInfo={userInfo.defaultOperator}
      />
      <TextInput
        id="medical-insurance-name"
        label="Convênio"
        hasDoubleContainer={true}
      />
    </form>
  );
};

export default PatientInfo;
