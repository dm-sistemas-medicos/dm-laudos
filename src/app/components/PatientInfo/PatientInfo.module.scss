.patient-info {
  padding: 4px;
  display: grid;
  grid-template-columns: 1fr auto auto;
  gap: 8px;
  font-size: 12px;
  border: solid #000 1px;

  > div {
    display: grid;
    grid-template-columns: auto 1fr;
    grid-auto-flow: column;
    align-items: center;
    column-gap: 2px;

    > input {
      font-size: 12px;
    }
  }
}

.extra-info {
  grid-column: span 2;
  grid-template-columns: auto auto 1fr;
}

.small-input {
  width: 48px;
}

.medium-input {
  width: 96px;
}

.full-input {
  grid-column: span 2;
}
