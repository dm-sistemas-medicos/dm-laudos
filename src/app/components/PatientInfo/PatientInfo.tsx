"use client";

import { useState } from "react";

import TextInput from "./components/TextInput/TextInput";
import BirthdateInput from "./components/BirthdateInput/BirthdateInput";
import MedicalReportDateInput from "./components/MedicalReportDateInput/MedicalReportDateInput";

import styles from "./PatientInfo.module.scss";

const today = new Date();

const PatientInfo = () => {
  const [medicalReportDate, setMedicalReportDate] = useState(
    today.toISOString().substring(0, 10)
  );

  return (
    <form
      className={styles["patient-info"]}
      data-testid="patient-info-container"
    >
      <TextInput id="name" label="Nome" />
      <TextInput
        id="medicalRecordNumber"
        label="Prontuário"
        isSmallInput={true}
      />
      <MedicalReportDateInput
        medicalReportDate={medicalReportDate}
        setMedicalReportDate={setMedicalReportDate}
      />
      <TextInput
        id="requesterName"
        label="Solicitante"
        initialInfo="DRA MARIANA MACHOSKI"
      />
      <BirthdateInput medicalReportDate={medicalReportDate} />
      <TextInput
        id="operatorName"
        label="Operador"
        initialInfo="DRA MARIANA MACHOSKI"
      />
      <TextInput
        id="medicalInsuranceName"
        label="Convênio"
        hasDoubleContainer={true}
      />
    </form>
  );
};

export default PatientInfo;
