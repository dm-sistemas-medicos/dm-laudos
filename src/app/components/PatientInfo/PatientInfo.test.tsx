import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import PatientInfo from "./PatientInfo";

describe("PatientInfo component", () => {
  beforeEach(() => {
    render(<PatientInfo />);
  });

  it("renders the component", () => {
    const patientInfo = screen.getByTestId("patient-info-container");

    expect(patientInfo).toBeInTheDocument();
  });

  it("renders the name info", () => {
    const name = screen.getByTestId("name-container");

    expect(name).toBeInTheDocument();
  });

  it("renders the medical-record-number info", () => {
    const medicalRecordNumber = screen.getByTestId(
      "medical-record-number-container"
    );

    expect(medicalRecordNumber).toBeInTheDocument();
  });

  it("renders the medical-record-date info", () => {
    const medicalReportDate = screen.getByTestId(
      "medical-report-date-container"
    );

    expect(medicalReportDate).toBeInTheDocument();
  });

  it("renders the requester-name info", () => {
    const requesterName = screen.getByTestId("requester-name-container");

    expect(requesterName).toBeInTheDocument();
  });

  it("renders the birthdate info", () => {
    const birthdate = screen.getByTestId("birthdate-input-container");

    expect(birthdate).toBeInTheDocument();
  });

  it("renders the operator-name info", () => {
    const operatorName = screen.getByTestId("operator-name-container");

    expect(operatorName).toBeInTheDocument();
  });

  it("renders the medical-insurance-name info", () => {
    const medicalInsuranceName = screen.getByTestId(
      "medical-insurance-name-container"
    );

    expect(medicalInsuranceName).toBeInTheDocument();
  });
});
