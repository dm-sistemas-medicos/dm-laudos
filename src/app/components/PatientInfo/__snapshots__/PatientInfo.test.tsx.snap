// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientInfo component renders the component 1`] = `
<div>
  <form
    class="patient-info"
  >
    <div
      data-testid="name-container"
    >
      <label
        for="name"
      >
        Nome
        :
      </label>
      <input
        data-testid="name"
        id="name"
        type="text"
        value=""
      />
    </div>
    <div
      data-testid="medical-record-number-container"
    >
      <label
        for="medical-record-number"
      >
        Prontuário
        :
      </label>
      <input
        class="small-input"
        data-testid="medical-record-number"
        id="medical-record-number"
        type="text"
        value=""
      />
    </div>
    <div
      data-testid="medical-report-date-container"
    >
      <label
        for="medical-report-date"
      >
        Data:
      </label>
      <input
        class="input"
        data-testid="medical-report-date"
        id="medical-report-date"
        type="date"
        value="2024-08-07"
      />
    </div>
    <div
      data-testid="requester-name-container"
    >
      <label
        for="requester-name"
      >
        Solicitante
        :
      </label>
      <input
        data-testid="requester-name"
        id="requester-name"
        type="text"
        value="DRA MARIANA MACHOSKI"
      />
    </div>
    <div
      class="container"
      data-testid="birthdate-input-container"
    >
      <label
        for="birthdate"
      >
        Nascimento:
      </label>
      <input
        class="input"
        data-testid="birthdate-input"
        id="birthdate"
        type="date"
        value=""
      />
      <span
        data-testid="lifetime-span"
      />
    </div>
    <div
      data-testid="operator-name-container"
    >
      <label
        for="operator-name"
      >
        Operador
        :
      </label>
      <input
        data-testid="operator-name"
        id="operator-name"
        type="text"
        value="DRA MARIANA MACHOSKI"
      />
    </div>
    <div
      class="double-container"
      data-testid="medical-insurance-name-container"
    >
      <label
        for="medical-insurance-name"
      >
        Convênio
        :
      </label>
      <input
        data-testid="medical-insurance-name"
        id="medical-insurance-name"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
